rocketpool-cli/node/sign-message.go

```diff
@@ -0,0 +1,57 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/urfave/cli"
+
+	"github.com/rocket-pool/smartnode/shared/services/rocketpool"
+	cliutils "github.com/rocket-pool/smartnode/shared/utils/cli"
+)
+
+func signMessage(c *cli.Context) error {
+
+	// Get RP client
+	rp, err := rocketpool.NewClientFromCtx(c)
+	if err != nil {
+		return err
+	}
+	defer rp.Close()
+
+	// Get & check wallet status
+	status, err := rp.WalletStatus()
+	if err != nil {
+		return err
+	}
+
+	if !status.WalletInitialized {
+		fmt.Println("The node wallet is not initialized.")
+		return nil
+	}
+
+	message := c.String("message")
+	if message == "" {
+		message = cliutils.Prompt("Please enter the message you want to sign: (EIP-191 personal_sign)", "^.+$", "Please enter the message you want to sign: (EIP-191 personal_sign)")
+	}
+
+	response, err := rp.SignMessage(message)
+	if err != nil {
+		return err
+	}
+
+	// Print the signature
+	fmt.Printf("Message: %s\n", message)
+	fmt.Printf("Signed data: %s\n\n", response.SignedData)
+
+	if cliutils.Confirm("Do you want to use this message on beaconcha.in? (Prints the message in the expected format)") {
+		fmt.Printf(`{ 
+    "address": "%s",
+    "msg": "%s",
+    "sig": "%s",
+    "version": "2"
+}`, status.AccountAddress, message, response.SignedData)
```

Comment:
make a struct and use decorators?

Reply:
Not sure I see the benefits. Could you elaborate if you think that's really important?
Btw, I'm changing the version to 1, meaning it's our first signature "spec". The 2 came from what's used by MyCripto and Beaconcha.in. But it can be anything.

Reply:
Sure. There's 2 advantages, really:

1. Indentation doesn't break, which is admittedly a nit
2. sprintf is much more error prone and less future proof. This is valid JSON... it's easy to look at it and see that. But if it ever gets more complex, we may miss a stray comma or something. golang's native serializer won't.

Reply:
Basically, something like this:

https://go.dev/play/p/vqE5ALXgVvI

Reply:
Let me argue a bit:
- Printing as you suggest breaks the identation, making it a single line. I personally prefer 
this (current)
{
   "address":"0x17...",
   "msg":"aasdçflkajdfçlakdsjfçalskdfdkjçalsdkfjaçldfkjaçldfkjaçdlfkajdçflkajdfçlakdfjçlsdkfjaçdslkfjaçdlfkjaçdflkajsdçflkajdsf",
   "sig":"0xd1b...",
    "version":"1"
}

over this, specially with big signatures
{"address":"0x17...","msg":"aasdçflkajdfçlakdsjfçalskdfdkjçalsdkfjaçldfkjaçldfkjaçdlfkajdçflkajdfçlakdfjçlsdkfjaçdslkfjaçdlfkjaçdflkajsdçflkajdsf","sig":"0xd1b1...","version":"1"}
- We have an api response struct that I thought about using for this purpose but we have some extra fields, like "error" that don't need to be printed. Would be weird to create another struct and transfer from one to the other just for printing purposes;
- The version on the struct would be of type uint, but beaconcha.in expects a string, so more steps to prepare the struct for printing;
- It's a very simple json and I don't see it growing complex.

Reply:
You can pretty-print the json with MarshalIndent: https://go.dev/play/p/YdU8c7uBb2c

I agree that printing extra fields feels weird, but I disagree that there's such a thing as a struct too small to bother with. As for not wanting to transfer fields... you already are transferring them into the variadric args of Printf. Transferring them to a new struct is pretty much the same.

Make Version a string if that's what beaconcha.in expects, of course.

Reply:
Agreed. There you go Patches.